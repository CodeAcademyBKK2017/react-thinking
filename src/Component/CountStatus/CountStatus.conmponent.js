import React, {Component} from 'react';
import PropTypes from 'prop-types';

class CountStatus extends Component {
  render () {
    return (
      <input type="text" value={this.props.value}/>
    );
  }
}
  
CountStatus.propTypes = {
  value: PropTypes.number.isRequired,
};

CountStatus.defaultProps = {
  value: '',
};

export default CountStatus;