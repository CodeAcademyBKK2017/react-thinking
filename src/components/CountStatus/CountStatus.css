.CountStatus {
    display: block;
    
    width: 240px;
    height: 180px;

    margin: 0 auto 50px;

    font-size: 48px;
    font-weight: bold;
    color: black;
    border: 3px solid black;
    background-color: #33b5ff;
    text-align: center;
    line-height: 180px;
}