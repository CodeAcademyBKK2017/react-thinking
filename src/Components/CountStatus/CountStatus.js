import React, { Component } from 'react';
import './CountStatus.css';

class CountStatus extends Component {

  render() {
    // const count = this.props.count;
    const {count} = this.props;
      
    return (
      <div className="countstatus">{count}</div>
    );
  }
}

export default CountStatus;