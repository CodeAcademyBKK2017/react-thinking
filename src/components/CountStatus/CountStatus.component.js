import React, { Component } from 'react';
import PropTypes  from 'prop-types';
import './CountStatus.css';

class CountStatus extends Component {
    render() {
        return (
            <input className="CountStatus" type="text" value={this.props.value} />
        );
    }
}

CountStatus.propTypes = {
    value: PropTypes.number.isRequired,
}

CountStatus.defaultProps = {
    value: 0,
}

export default CountStatus;
