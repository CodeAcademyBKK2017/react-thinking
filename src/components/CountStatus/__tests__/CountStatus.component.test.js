import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme, {shallow} from 'enzyme';
import CountStatus from '../CountStatus.component';

describe('CountStatus', () => {
  it('Overlay: renders correctly', () => {
    const tree = renderer.create(<CountStatus />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});