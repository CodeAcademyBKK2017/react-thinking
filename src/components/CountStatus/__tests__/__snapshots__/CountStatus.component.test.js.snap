// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CountStatus Overlay: renders correctly 1`] = `
<input
  className="CountStatus"
  type="text"
  value={0}
/>
`;
